fendermint/app/src/app.rs

```diff
@@ -620,11 +623,42 @@ where
         *guard = Some(state);
 
         let response = match result {
-            Err(e) => invalid_check_tx(AppError::InvalidEncoding, e.description),
+            Err(e) => {
```

Comment:
Oh dear, these recursive results are such a telling symptom of just how unergonomic the nested interpreter design is /tech_debt 😢

Reply:
Yes indeed. I think it might be also worth to parse the transaction directly here - with help of external library.